scripts/common_cmdline.sh

```diff
@@ -71,20 +75,31 @@ function compileFull()
     set -e
     rm "$stderr_path"
 
-    if [[ \
-        ("$exit_code" -ne "$expected_exit_code" || \
-            ( $expect_output -eq 0 && -n "$errors" ) || \
-            ( $expect_output -ne 0 && $expected_exit_code -eq 0 && $expect_output -ne 2 && -z "$errors" ))
+    if [[
+        "$exit_code" != "$expected_exit_code" ||
+        "$errors" != "" && "$expect_output" == 'none' ||
+        "$errors" == "" && "$expect_output" != 'none' && "$expect_output" != 'any' && "$expected_exit_code" == 0
```

Comment:
You only really need quotes on the right side of operators inside `[[ ]]` (because otherwise Bash will treat them as regexes). Left side can remain unquoted which somewhat reduces the visual noise.